web/src/lib/stores/search-params.store.ts

```diff
@@ -0,0 +1,61 @@
+import { goto } from '$app/navigation';
+import { page } from '$app/stores';
+import { writable, get, derived, readonly, type Writable } from 'svelte/store';
+
+// TODO re-visit the types.
+const searchParams = new Map<string, unknown>();
+
+export class SearchParams<T extends string> {
+  private key: string;
+  private store$: Writable<Array<T>>;
+
+  private constructor(key: string) {
+    this.key = key;
+    this.store$ = writable<Array<T>>((get(page).url.searchParams.get(this.key)?.split(',') || []) as Array<T>);
+    this.store$.subscribe(this.handleUpdate);
+  }
+
+  private handleUpdate = (values: Array<T>) => {
+    if (values.length === 0) {
+      get(page).url.searchParams.delete(this.key);
+    } else {
+      get(page).url.searchParams.set(this.key, values.join(','));
+    }
+    goto(`?${get(page).url.searchParams.toString()}`);
```

Comment:
It's probably better to have something like this : 

```ts
const newUrl = new URL($page.url);
if (values.length === 0) {
  newUrl.searchParams.delete(this.key);
} else {
  newUrl.url.searchParams.set(this.key, values.join(','));
}
goto(newUrl);
```

based on this https://github.com/sveltejs/kit/issues/2785#issuecomment-967790973

I'm planning to refactor across the entire webapp